### PR TITLE
v6.16.1

### PR DESCRIPTION
**Pull request by cherniavskii**
- [x] Compare the last tag with the branch upon which you want to release (`next` for the alpha / beta releases and `master` for the current stable version).
      To do so, use `yarn release:changelog` The options are the following:

  ```bash
  yarn release:changelog
     --githubToken   YOUR_GITHUB_TOKEN (needs "public_repo" permission)
     --lastRelease   The release to compare against (default: the last one)
     --release       The branch to release (default: master)
  ```

  You can also provide the github token by setting `process.env.GITHUB_TOKEN` variable.

  In case of a problem, another method to generate the changelog is available at the end of this page.

- [x] Clean the generated changelog, to match the format of [https://github.com/mui/mui-x/releases](https://github.com/mui/mui-x/releases).
- [x] Update the root `package.json`'s version
- [x] Update the versions of the other `package.json` files and of the dependencies with `yarn release:version` (`yarn release:version prerelease` for alpha / beta releases).
- [ ] Open PR with changes and wait for review and green CI.
- [ ] Merge PR once CI is green, and it has been approved.

### Release the packages

- [ ] Checkout the last version of the working branch
- [ ] Make sure you have the latest dependencies installed: `yarn`.
- [ ] Build the packages: `yarn release:build`.
- [ ] Release the versions on npm: `yarn release:publish` (you need your 2FA device).
- [ ] Push the newly created tag: `yarn release:tag`.

### Publish the documentation

The documentation must be updated on the `docs-vX` branch (`docs-v4` for `v4.X` releases, `docs-v5` for `v5.X` releases, ...)

- [ ] Push the working branch on the documentation release branch to deploy the documentation with the latest changes.

```bash
git push upstream master:docs-v6 -f
```

You can follow the deployment process [on the Netlify Dashboard](https://app.netlify.com/sites/material-ui-x/deploys?filter=docs-v6)
Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for the `docs-v6` deployment.

### Publish GitHub release

- [ ] After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)

### Announce

- [ ] Follow the instructions in https://mui-org.notion.site/Releases-7490ef9581b4447ebdbf86b13164272d.

### CODE REVIEW

File: packages/x-codemod/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-codemod",
-  "version": "6.9.0",
+  "version": "6.16.1",
```
Comment (cherniavskii) [inline packages/x-codemod/package.json:3]: Caused by dependencies update

File: packages/x-license-pro/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-license-pro",
-  "version": "6.10.2",
+  "version": "6.16.1",
```
Comment (cherniavskii) [inline packages/x-license-pro/package.json:3]: Caused by dependencies update

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,6 +3,68 @@
...
 See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.
 
+## 6.16.1
+
+_Oct 5, 2023_
+
+We'd like to offer a big thanks to the 10 contributors who made this release possible. Here are some highlights ✨:
+
+- 🐞 Bugfixes
+- 📚 Documentation improvements
```
Comment (alexfauquette) [inline CHANGELOG.md:13]:
```suggestion
- 📚 Documentation improvements
- 🥧 Add interaction on pie chart
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,6 +3,68 @@
...
+#### `@mui/x-date-pickers-pro@6.16.1` [![pro](https://mui.com/r/x-pro-svg)](https://mui.com/r/x-pro-svg-link 'Pro plan')
+
+Same changes as in `@mui/x-date-pickers@6.16.1`, plus:
+
+- [DateRangePicker] Fix `InputProps` propagation in multi input (#10564) @alexfauquette
+
+### Charts / `@mui/x-charts@6.0.0-alpha.14`
+
+- [charts] Display cursor pointer for pie chart only if `onClick` is provided (#10551) @giladappsforce
+- [charts] PieChart onClick prop in the series data and demo (#10506) @giladappsforce
```
Comment (alexfauquette) [inline CHANGELOG.md:48]:
```suggestion
- [charts] Add `onClick` prop to PieChart (#10506) @giladappsforce
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,6 +3,68 @@
...
+### Tree View / `@mui/x-tree-view@6.0.0-beta.1`
+
+- Update dependencies
+
+### Docs
+
+- [docs] Add `DateRangePicker` example with a `Button` trigger (#10485) @LukasTy
+- [docs] Add section about disabling columns panel (#10328) @MBilalShafi
+- [docs] Add section about overriding slots to base concepts (#10421) @noraleonte
+- [docs] Add whats new page listing all release announcements (#9727) @joserodolfofreitas
```
Comment (alexfauquette) [inline CHANGELOG.md:59]:
```suggestion
- [docs] Add "whats new" page listing all release announcements (#9727) @joserodolfofreitas
```

**Review by alexfauquette — APPROVED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,6 +3,68 @@
...
+
+Same changes as in `@mui/x-date-pickers@6.16.1`, plus:
+
+- [DateRangePicker] Fix `InputProps` propagation in multi input (#10564) @alexfauquette
+
+### Charts / `@mui/x-charts@6.0.0-alpha.14`
+
+- [charts] Display cursor pointer for pie chart only if `onClick` is provided (#10551) @giladappsforce
+- [charts] PieChart onClick prop in the series data and demo (#10506) @giladappsforce
+
```
Comment (alexfauquette) [inline CHANGELOG.md:49]:
```suggestion
- [charts] Support `slots`/`slotProps` for the tooltip (#10515) @alexfauquette

```

File: CHANGELOG.md [hunk 7]
```diff
@@ -3,6 +3,68 @@
...
+- [docs] Add `DateRangePicker` example with a `Button` trigger (#10485) @LukasTy
+- [docs] Add section about disabling columns panel (#10328) @MBilalShafi
+- [docs] Add section about overriding slots to base concepts (#10421) @noraleonte
+- [docs] Add whats new page listing all release announcements (#9727) @joserodolfofreitas
+- [docs] Update RTL Support section of the grid localization docs (#10561) @MBilalShafi
+
+### Core
+
+- [core] Consistent casing with Legal and marketing content @oliviertassinari
```
Comment (LukasTy) [inline CHANGELOG.md:64]:
```suggestion
- [core] Fix casing consistency with legal and marketing content @oliviertassinari
```

File: packages/x-codemod/package.json (hunk 1 above)
Comment (cherniavskii) [reply packages/x-codemod/package.json:3]: Good point, I think it makes sense to skip given that upgraded dependencies are not pinned to specific versions (they use range version), so the users can upgrade `@mui/x-license-pro`'s dependencies even if we don't release the package.

Comment (LukasTy) [reply packages/x-codemod/package.json]: Yes, that's my point. 💡 
If we used direct dependency versions, then sure, we would basically be forced to release every time. 🤔 👌

File: packages/x-codemod/package.json [hunk 8]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-codemod",
-  "version": "6.9.0",
+  "version": "6.16.0",
```
Comment (LukasTy) [inline packages/x-codemod/package.json:3]:
```suggestion
  "version": "6.9.0",
```

File: packages/x-license-pro/package.json [hunk 9]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@mui/x-license-pro",
-  "version": "6.10.2",
+  "version": "6.16.0",
```
Comment (LukasTy) [inline packages/x-license-pro/package.json:3]:
```suggestion
  "version": "6.10.2",
```

**Review by michelengelen — APPROVED** [review]
LGTM :shipit:

**Review by LukasTy — APPROVED** [review]

**Pull request merged**